### PR TITLE
Fix ConfigObject warnings and issues

### PR DESCRIPTION
**Pull request by uklotzde**
- Fix GCC 9 warnings
- Inline pass-through constructors
- Add virtual destructor to ConfigValue
- Define both operator == and !=
- Delete unused functions
- Add TODO to fix the error-prone design of ConfigValueKbd

### CODE REVIEW

**Comment by Be-ing** [comment]
I have changed the target branch to master. Unless there is some bug that impacts users that you have identified, I do not think this warrants changing the stable branch. Cleaning up code is not a good reason to risk regressions on a stable branch.

**Comment by uklotzde** [comment]
@Be-ing Agreed.

File: src/preferences/configobject.cpp [hunk 1]
```diff
@@ -117,37 +96,14 @@ ConfigValue::ConfigValue(double dValue)
...
-    return (s1.value.toUpper() == s2.value.toUpper());
+ConfigValueKbd::ConfigValueKbd(QString value)
+        : ConfigValue(std::move(value)),
+          m_keys(ConfigValue::value) {
 }
 
-bool operator==(const ConfigValueKbd& s1, const ConfigValueKbd& s2) {
-    //qDebug() << s1.m_qKey << "==" << s2.m_qKey;
-    return (s1.m_qKey == s2.m_qKey);
+ConfigValueKbd::ConfigValueKbd(QKeySequence keys)
```
Comment (daschuer) [inline src/preferences/configobject.cpp]: QKeySequence

**Review by daschuer — CHANGES_REQUESTED** [review]

**Comment by daschuer** [comment]
LGTM. Thank you. The build errors are unrelated.

**Pull request merged**